Doc/library/ast.rst

```diff
@@ -146,6 +146,98 @@ Node classes
     Snakes <https://greentreesnakes.readthedocs.io/en/latest/>`__ project and
     all its contributors.
 
+
+.. _ast-root-nodes:
+
+Root nodes
+^^^^^^^^^^
+
+.. class:: Module(body, type_ignores)
+
+   A Python module.
+
+   *body* is a :class:`list` of the module's :ref:`ast-statements`.
+
+   *type_ignores* is a :class:`list` of the module's type ignore comments;
+   see :func:`ast.parse` for more details.
+
+   .. doctest::
+
+        >>> print(ast.dump(ast.parse('x = 1'), indent=4))
+        Module(
+            body=[
+                Assign(
+                    targets=[
+                        Name(id='x', ctx=Store())],
+                    value=Constant(value=1))],
+            type_ignores=[])
+
+
+.. class:: Expression(body)
+
+   A single Python expression.
+
+   *body* is a single node,
+   one of the :ref:`expression types <ast-expressions>`.
+
+   .. doctest::
+
+        >>> print(ast.dump(ast.parse('123', mode='eval'), indent=4))
+        Expression(
+            body=Constant(value=123))
+
+
+.. class:: Interactive(body)
+
+   A single Python interactive line, like in :ref:`tut-interac`.
```

Comment:
This doesn't necessarily mean a line, but rather a single interactive input. Not sure if we have an official term for it, but I think we can use the same description as the `compile()`:

```suggestion
   A single interactive statement, like in :ref:`tut-interac`.
```

Reply:
IIRC, it originally read as in your suggestion, but I found it rather confusing and contradictory, because (as stated on the very next line):

> *body* is a `list` of statement nodes

We're simultaneously saying that an `Interactive` node instance is "a _single_ interactive statement" and (composed of) "a list of statement nodes", which is naturally confusing and seemingly contradictory.

As I understand it, it is the former that is not really correct, because an interactive input can consist of an arbitrary number of statements separated by `;`, and `ast.parse` will handle this accordingly, e.g.

```python
>>> ast.parse("a = 1; b = 2", mode="single")
<ast.Interactive object at 0x00000000028A92D0>
>>> ast.parse("a = 1; b = 2", mode="single").__dict__
{'body': [<ast.Assign object at 0x00000000036D6560>, <ast.Assign object at 0x00000000036D67D0>]}
```

Therefore, unless someone comes up with a better word, it would seem the best we can do is:

```suggestion
   A single interactive input, like in :ref:`tut-interac`.
```

Reply:
> As I understand it, it is the former that is not really correct, because an interactive input can consist of an arbitrary number of statements separated by ;, and ast.parse will handle this accordingly, e.g.

This is why I preferred to use "interactive statement" rather than "a statement". As far as I can tell, that's the official name we have been using for the `compile()` function.

> As I understand it, it is the former that is not really correct, because an interactive input can consist of an arbitrary number of statements separated by ;, and ast.parse will handle this accordingly, e.g.

An arbitrary number of 'simple statements' (e.g. statements without any sort of nesting / body block) or a single compound statement. So if we go into the details it becomes even more complex to describe.
https://github.com/python/cpython/blob/b53bad6dd08db78f5b5e2f41bea69c4b657fca13/Grammar/python.gram#L101-L105

In any case, I think I'd be fine with `interactive input` since that is still more cleaner than a single line (which is untrue).

Reply:
> This is why I preferred to use "interactive statement" rather than "a statement". As far as I can tell, that's the official name we have been using for the compile() function.

Hmm, I see. Still, though, it seems likely that readers would not understand "interactive statement" as a distinct term that means something more specific than just "a statement at the interactive prompt" (as I did), unless it is at minimum marked up and actually defined/linked to a definition. Besides single usage of the phrase "interactive statement" in the `compile()` function doc, as you mention, in the description of `"single"` argument to the `mode` parameter (which doesn't define it), I couldn't find any other mentions, much less definitions of that specific term in the Python docs either with the built-in search or Google. Thus, overloaded as it is, it reads as a _prima facie_ contradiction with the next sentence, with obvious confusion potential.

> An arbitrary number of 'simple statements' (e.g. statements without any sort of nesting / body block) or a single compound statement. So if we go into the details it becomes even more complex to describe.

Ah right, of course—thanks :+1: 

> In any case, I think I'd be fine with `interactive input` since that is still more cleaner than a single line (which is untrue).

Indeed, and while I hadn't found it before, "interactive input" is in fact the term used for this [in the language reference](https://docs.python.org/3.12/reference/toplevel_components.html#interactive-input):

> ## 9.3. Interactive input
>
> Input in interactive mode is parsed using the following grammar:
>
> ```ebnf
> interactive_input ::= [ stmt_list ] NEWLINE | compound_stmt NEWLINE
> ```

Reply:
Ah, that's a very nice find indeed! I'm sold on "interactive input", let's go for it!